in_toto/models/gpg.py

```diff
@@ -0,0 +1,250 @@
+    hashes: list of GPG Hash Algorithms, e.g. "pgp+SHA2".
+    keyval: Opaque key content.
+    keyid: Key identifier that is unique within the metadata it is used in.
+      Keyid is not verified to be the hash of a specific representation
+      of the key.
+    creation_time: Unix timestamp when GPG key was created.
+    validity_period: Validity of the GPG Keys in days.
+    subkeys: A dictionary containing keyid and GPG subkey.
+  """
+
+  type: str
+  method: str
+  hashes: List[str]
+  keyval: Dict[str, str]
+  keyid: str
+  creation_time: Optional[int] = None
+  validity_period: Optional[int] = None
+  subkeys: Optional[Dict[str, "_LegacyGPGKey"]] = None
+
+  @classmethod
+  def from_dict(cls, keyid: str, key_dict: Dict[str, Any]):
+    """Creates ``_LegacyGPGKey`` object from its json/dict representation.
+    Raises:
+      KeyError, TypeError: Invalid arguments.
+    """
+    subkeys_dict = key_dict.get("subkeys")
+
+    gpg_subkeys = None
+    if subkeys_dict:
+      gpg_subkeys = {
+        _keyid: _LegacyGPGKey.from_dict(_keyid, subkey_dict)
+        for (_keyid, subkey_dict) in subkeys_dict.items()
+      }
+
+    return cls(
+      key_dict["type"],
+      key_dict["method"],
+      key_dict["hashes"],
+      key_dict["keyval"],
+      keyid,
+      key_dict.get("creation_time"),
+      key_dict.get("validity_period"),
+      gpg_subkeys,
+    )
+
+  @classmethod
+  def from_legacy_dict(cls, key_dict: Dict[str, Any]):
+    """Create GPGKey from legacy dictionary representation."""
+
+    keyid = key_dict["keyid"]
+    return cls.from_dict(keyid, key_dict)
+
+  def to_dict(self):
+    """Returns the dictionary representation of self."""
+
+    key_dict = {
+      "method": self.method,
+      "type": self.type,
+      "hashes": self.hashes,
+      "keyid": self.keyid,
+      "keyval": self.keyval,
+    }
+
+    if self.creation_time:
+      key_dict["creation_time"] = self.creation_time
+    if self.validity_period:
+      key_dict["validity_period"] = self.validity_period
+    if self.subkeys:
+      subkeys_dict = {
+        keyid: subkey.to_dict()
+        for (keyid, subkey) in self.subkeys.items()
+      }
+      key_dict["subkeys"] = subkeys_dict
+
+    return key_dict
+
+  @classmethod
+  def from_keyring(cls, keyid, homedir=None):
+    """Creates ``_LegacyGPGKey`` object from GnuPG Keyring."""
+
+    pubkey_dict = gpg.export_pubkey(keyid, homedir)
+    return cls.from_dict(keyid, pubkey_dict)
+
+  def verify_signature(
+    self,
+    signature: _LegacyGPGSignature,
+    data: bytes
+  ) -> bool:
+    """Verifies a given payload by the key assigned to the _LegacyGPGKey
+    instance.
+
+    Arguments:
+      signature: A ``_LegacyGPGSignature`` class instance.
+      payload: The bytes to be verified.
+    Returns:
+      Boolean. True if the signature is valid, False otherwise.
+    """
+
+    return gpg.verify_signature(signature.to_dict(), self.to_dict(), data)
```

Comment:
Can you take a look at `verify_signature` functions in securesystemslib? They should raise or return None (see e.g. https://github.com/secure-systems-lab/securesystemslib/pull/488)